dbt/adapters/spark/impl.py

```diff
@@ -90,53 +107,79 @@ def drop_relation(self, relation, model_name=None):
             kwargs={'relation': relation}
         )
 
-    def get_catalog(self, manifest):
-        schemas = manifest.get_used_schemas()
+    @staticmethod
+    def _parse_relation(relation: Relation,
```

Comment:
There's a lot going on in this method! Can we break this into two (or more) methods that are a little more self-contained?

We should have at least one method, `parse_describe_extended_results`, or similar. This method takes the agate table response from `describe extended...` and returns a data structure containing a list of columns, plus an arbitrary key/value dict of properties returned by the statement. From there, other methods can operate on a proper list of Column objects, plus any properties parsed out from the `describe extended` call as needed.